api/hasura/actions/createUser.ts

```diff
@@ -42,6 +42,25 @@ async function handler(req: VercelRequest, res: VercelResponse) {
     });
   }
 
+  // make sure a profile exists first, for FK purposes
+
+  const { profiles } = await adminClient.query({
```

Comment:
you should do an upsert here to save 1 less query

```gql
const { insert_profiles_one } = await adminClient.mutate({
        // Create a profile if none exists yet
        insert_profiles_one: [
          {
            object: { address: input.address },

            // This clause allows gql to catch the conflict and do nothing
            // hasura calls this an "upsert"
            on_conflict: {
              constraint: profiles_constraint.profiles_address_key,
              // Don't update the entry at all if a profile exists
              // Don't want to touch the timestamp if we aren't actually
              // modifying anything
              update_columns: [],
            },
          },
          {
            id: true,
          },
        ],
      });
```

Reply:
@zashton  This doesnt handle the ilike case